File: api/BeHealth.Business/Resources/FoodManamgnet/Templates/MealResource.cs
using BeHealth.Business.Resources.FoodManamgnet.Meals;
using System;
using System.Collections.Generic;
using System.Text;

namespace BeHealth.Business.Resources.FoodManamgnet.Templates
{
    public class MealResource
    {
        public long ID { get; set; }

        public string Name { get; set; }

        public DateTimeOffset Time { get; set; }

        public List<RecipeResource> Recipes { get; set; }
        public List<MealFoodItemResource> Items { get; set; }

        public double Fat { get; set; }
        public double Energy { get; set; }
        public double Protien { get; set; }
        public double Carbs { get; set; }
    }
}

File: app/src/app/features/templates/resources/meal.resource.ts
import { RecipeResource } from '../../recipes/resources/recipe.resource';
import { MealFoodItemResource } from './meal-food-item.resource';

export class MealResource {
  id: number;
  name: string;
  time: Date;
  recipes: Array<RecipeResource>;
  items: Array<MealFoodItemResource>;
  fat: number;
  energy: number;
  protien: number;
  carbs: number;
}
